### PR TITLE
[ll][gl] glutin integration

### PR DESCRIPTION
**Pull request by kvark**
A follow-up to #1468
I think `gl` needs an `Instance` type, and every other backend needs, too.
Perhaps, the `Instance` should be implementing the `Backend` trait even, to make the passing around and initialization more consistent.

### CODE REVIEW

File: src/backend/gl/Cargo.toml [hunk 1]
```diff
@@ -12,10 +12,15 @@ documentation = "https://docs.rs/gfx_device_gl"
 workspace = "../../.."
 
 [lib]
-name = "gfx_device_gl"
+name = "gfx_backend_gl"
+
+[features]
+default = ["glutin"]
+headless = ["glutin"]
```
Comment (msiglreith) [inline src/backend/gl/Cargo.toml]: I think we can remove the `headless` feature as it got promoted to a first-class feature with the introduction of `Headless` trait.

File: src/backend/gl/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
-name = "gfx_device_gl"
-version = "0.14.0"
+name = "gfx_backend_gl"
```
Comment (Bastacyclop) [inline src/backend/gl/Cargo.toml]: Nice to see it renamed! :D Although a crates.io search for "gfx" might look weird.. Is there a way to remove a crate from public search like a full crate yank ?

Comment (msiglreith) [reply src/backend/gl/Cargo.toml]: I would prefer a large `[DEPRECATED]` sign in the crate description text for the `gfx_device_` and `gfx_window_` crates.

**Comment by kvark** [comment]
@msiglreith PTAL, I think this is ready to merge now

File: src/backend/gl/src/window/glutin.rs [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        };
+
+        Swapchain {
+            window: self.window.clone(),
+            backbuffer: [backbuffer; 1],
+        }
+    }
+}
+
+impl core::Instance<B> for Surface {
```
Comment (msiglreith) [inline src/backend/gl/src/window/glutin.rs]: ha! That's an interesting approach :D

**Pull request merged**